### ISSUE TITLE
NoPCM SRS assumes an implementation choice?

### ISSUE THREAD

**Issue by bmaclach** [JacquesCarette/Drasil#1498]
It's been mentioned in another issue (#492) how the differential equations in NoPCM have an analytical solution, and thus do not necessarily require an ODE solver. However, we still have the `abs_tol`, `rel_tol`, and `timeStep` inputs in the NoPCM SRS, which are only used as parameters for an ODE solver. By including these as inputs, are we not making a decision about the implementation (choosing to use an ODE solver)? If we removed these inputs, would we then be making the opposite decision (choosing to use the analytical solution)? It seems almost inevitable that we need to make this decision in the requirements stage. Should we add to the "System Constraints" section a note that the solution will use an ODE solver?

**Comment by smiths** [comment]
Great observation @bmaclach.  You are absolutely correct that requiring, or not requiring, ODE solver parameters constrains the design.  Rather than constrain the design, the better course of action is to move these quantities to the design stage.  The SRS really should just be about the physics.  The design choices will add new requirements.  The problem for us is that we haven't yet figured out our design documentation.  If we make the design documentation the same style as the SRS, essentially requirements for the design, we could fit this in fairly naturally.  I think the best course of action is to really think about the Projectile example that @samm82 is working on.  This example seems like a good platform for experimentation.  It is complex enough, but still fairly simple.

If I've answered your question, you can close this issue.

**Comment by JacquesCarette** [comment]
I don't like the phrasing "design choices will add new requirements". I agree that design choices will add new information that needs to be considered properly and appropriately - though I strongly hesitate to call them requirements [I can be convinced otherwise though]. "requirements for the design" is closer, but awkward.

**Comment by smiths** [comment]
@JacquesCarette, I agree that saying "design choices add new requirements" isn't exactly what I wanted to say.  For all of SEs work on requirements and design, the terminology isn't quite as expressive as we want it to be.  :smile:  What I would like to see is the requirements document not mention anything about numerical parameters.  As soon as we do that, we constrain the design.  

When we make the design choices, within the design documentation, we have constrained the requirements.  We can maybe think of the design choices as software constraints.  That is the term I would use in the SRS if design decisions were externally imposed at the beginning of the process.

Somewhere we need to say that the choice to use an ODE solver has added a need for tolerances and a time step.  I think we say that in our design documentation (whatever that might look like) and then the code generator gets this information from here.  In my loose language above, I likened that to adding a "requirement."
